src/transports/ws/sws.c

```diff
@@ -773,14 +744,21 @@ static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
 
     self->fail_msg_len += payload_len;
 
-    self->instate = NN_SWS_INSTATE_FAILING;
+    self->instate = NN_SWS_INSTATE_CLOSED;
 
-    /*  On the next recv, the connection will be failed. Why defer
-        until the next recv? Semantically, until then, this incoming
-        message has not been interpreted, so it's not until then that
-        it could be failed. This type of pre-processing is necessary
-        to early fail chunked transfers. */
```

Comment:
It came about as I started looking at cleaning up changes to remove those functions and add the type option.  When I started testing I found that I got back a zero length frame which was NOT what should have happened.  And that led me here. 

So yes I encountered specific behavior that I believed was incorrect.  I realize you did it that way on purpose but I think you've since come to agree with me that websockets should not be a special snowflake but instead behave just like any other nanomsg transport. 

Sent from my iPhone

> On Oct 24, 2015, at 8:04 PM, Wirebird Labs LLC <notifications@github.com> wrote:
> 
> In src/transports/ws/sws.c:
> 
> >  
> > -    /*  On the next recv, the connection will be failed. Why defer
> > -        until the next recv? Semantically, until then, this incoming
> > -        message has not been interpreted, so it's not until then that
> > -        it could be failed. This type of pre-processing is necessary
> > -        to early fail chunked transfers. */
> A more direct way of talking about this modification -- did you change it in order to simplify what seemed "wonky" or "wrong", or was it specifically to address a characterized failure? Don't take this question as accusatory -- rather, it's highlighting my own immediate lack of ability to give positive, helpful review of this change, without yet having mentally spooled up again!
> 
> —
> Reply to this email directly or view it on GitHub.
>